spec/datadog/appsec/processor_spec.rb

```diff
@@ -0,0 +1,227 @@
+# typed: ignore
+
+require 'datadog/appsec/spec_helper'
+require 'datadog/appsec/processor'
+
+RSpec.describe Datadog::AppSec::Processor do
+  before do
+    skip 'disabled for Java' if RUBY_PLATFORM.include?('java')
+
+    logger = double(Datadog::Core::Logger)
+    allow(logger).to receive(:debug?).and_return true
+    allow(logger).to receive(:debug)
+    allow(logger).to receive(:info)
+    allow(logger).to receive(:warn)
+    allow(logger).to receive(:error)
+
+    allow(Datadog).to receive(:logger).and_return(logger)
+  end
+
+  context 'self' do
+    it 'detects if the WAF is unavailable' do
+      hide_const('Datadog::AppSec::WAF')
+
+      expect(described_class.libddwaf_required?).to be false
+    end
+
+    it 'detects if the WAF is available' do
+      stub_const('Datadog::AppSec::WAF', Module.new)
+
+      expect(described_class.libddwaf_required?).to be true
+    end
+
+    it 'reports via return of libddwaf loading failure' do
+      allow(Object).to receive(:require).with('libddwaf').and_raise(LoadError)
```

Comment:
I suggest using `allow(described_class).to receive(:require)` as mocking such an important method on `Object` may accidentally mean breaking rspec when it tries to report a failure.

(We've had issues like this in dd-trace-rb in specs that mocked very core functionality).